src/runtime/pprof/pprof_test.go

```diff
@@ -1230,6 +1230,50 @@ func TestMutexProfile(t *testing.T) {
 	})
 }
 
+func TestMutexProfileChangeRate(t *testing.T) {
+	old := runtime.SetMutexProfileFraction(1)
+	defer runtime.SetMutexProfileFraction(old)
+
+	countcontentions := func() int64 {
+		var w bytes.Buffer
+		Lookup("mutex").WriteTo(&w, 0)
+		p, err := profile.Parse(&w)
+		if err != nil {
+			t.Fatalf("failed to parse profile: %v", err)
+		}
+		t.Logf("parsed proto: %s", p)
+		if err := p.CheckValid(); err != nil {
+			t.Fatalf("invalid profile: %v", err)
+		}
+
+		var count int64
+		for _, s := range p.Sample {
+			count += s.Value[0]
+		}
+		return count
+	}
+
+	original := countcontentions()
+	newrate := 2
+	for i := 0; i < newrate*50; i++ {
+		blockMutex(t)
+	}
+	before := countcontentions() - original
+	runtime.SetMutexProfileFraction(newrate)
+	for i := 0; i < 50; i++ {
+		blockMutex(t)
+	}
+	after := countcontentions() - original
```

Comment:
🤔  I'm wondering if a simpler test is possible?

1. Set mutext profile fraction to X
2. Simulate N block events
3. Get `countcontentions()` A
4. Set mutext profile fraction to Y
5. Get `countcontentions()` B
6. Verify A == B

Seems like it's a bit simpler and has no chance to vary due to random sampling?

Reply:
I actually wrote that exact test originally, and it's sufficient to demonstrate the unintuitive behavior. The reason I went with this test is that I wanted something that demonstrated that changing the sampling rate actually had the expected effect when recording samples at different rates. I wanted to show that the overall profile didn't look completely unbelievable. But maybe my current test is more complicated than it's worth for this issue?

Reply:
I don't feel strongly. Ultimately it'll be up to the Go authors preferences 🙈.